jax_unirep/featurize.py

```diff
@@ -49,7 +48,7 @@ def rep_arbitrary_lengths(
     :param seqs: A list of sequences as strings.
         If passing only a single sequence, it also needs to be passed inside a list.
```

Comment:
Need to add docstrings for `apply_fun` and `mlstm_size`.

I proposed a docstring above for `apply_fun`, which can be reused.

Here's one for `mlstm_size`:

```
:param mlstm_size: Integer specifying the number of nodes in the mLSTM layer.
    Though the model architecture space is practically infinite,
    we assume that you are using the same number of nodes per mLSTM layer.
    (This is a common simplification used in the design of neural networks.)
```

Reply:
A second thought also came to mind: I do know that you did a lot of really clever work designing the reps of arbitrary lengths without padding, especially to get back the original ordering. I'm also recalling that I was originally hesitant to do padding because I wasn't sure what it would do to the sequence reps. Now, however, I think at least one user has shown us that using padding results in more training dynamics. What's your thoughts on tentatively marking `rep_arbitrary_lengths` as deprecated? Doing so means we can leave it in the codebase, but we also relieve the pressure of having to update two functions that do _almost_ the same thing but can't be easily combined, as marking it as deprecated means aren't committing to keeping the function updated.

Reply:
Hm, for evotuning, yes, padding seems to be the better choice often, since it makes the sequence batches more diverse. In repping though, you probably use the embeddings straight away for either downstream prediction or clustering, and there I think padding will introduce unnecessary noise. Or am I missing the point here?

Reply:
You're right! I keep forgetting about this point. 

Looks like we need to have documentation on why `reps_arbitrary_lengths` exists, if anything, to prevent myself from repeatedly making this suggestion in the future. It stems from the original workflow of "rep every sequence one at a time". We are speeding this particular workflow up by batching things together and repping things _together_, thereby avoiding Python loop penalties.

Can you add the following text in an appropriate place in the docstring?

```
This function exists to speed up the original published workflow of "repping one sequence at a time".
Repping one at a time avoids generating noise in the reps from adding padding characters.
```